Error building tensorflow-gpu

Hi, I get this error when building tensorflowWithCuda.

```
ERROR: /build/output/external/hwloc/BUILD.bazel:229:1: C++ compilation of rule '@hwloc//:hwloc' failed (Exit 1)
[7,212 / 12,420] 31 actions running
    Compiling external/mkl_dnn/src/cpu/cpu_engine.cpp; 42s local
    @mkl_dnn//:mkldnn_single_threaded; 42s local
    @mkl_dnn//:mkldnn_single_threaded; 28s local
    @mkl_dnn//:mkldnn_single_threaded; 25s local
    Compiling external/mkl_dnn/src/cpu/cpu_reorder.cpp; 23s local
    Compiling external/mkl_dnn/src/cpu/rnn/ref_rnn.cpp; 14s local
    Compiling .../xla/service/cpu/runtime_single_threaded_conv2d.cc; 9s local
    Compiling .../service/cpu/runtime_single_threaded_matmul.cc; 9s local ...
external/hwloc/hwloc/topology.c:45:10: fatal error: sys/sysctl.h: No such file or directory
 #include <sys/sysctl.h>
```

Steps to reproduce:

```
nix-shell -p python37Packages.tensorflowWithCuda
```

Same problem when trying to build `python36Packages.dm-sonnet` which depends on tensorflow:
```
ERROR: /build/output/external/hwloc/BUILD.bazel:229:1: C++ compilation of rule '@hwloc//:hwloc' failed (Exit 1)
[4,944 / 5,456] 7 actions running
    Compiling tensorflow/core/util/batch_util.cc [for host]; 24s local
    Compiling .../core/framework/graph_to_functiondef.cc [for host]; 4s local
    Compiling tensorflow/core/framework/dataset.cc [for host]; 2s local
    Compiling .../core/common_runtime/bfc_allocator.cc [for host]; 0s local
    Compiling tensorflow/core/framework/collective.cc [for host]; 0s local
    Compiling .../core/common_runtime/allocator_retry.cc [for host]; 0s local
    Compiling external/hwloc/hwloc/topology-linux.c [for host]; 0s local
external/hwloc/hwloc/topology.c:45:10: fatal error: sys/sysctl.h: No such file or directory
   45 | #include <sys/sysctl.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.
```

--- a/pkgs/development/python-modules/tensorflow/1/default.nix
+++ b/pkgs/development/python-modules/tensorflow/1/default.nix
@@ -258,6 +258,11 @@ let
 
       substituteInPlace tensorflow/tools/pip_package/setup.py \
         --replace "numpy >= 1.16.0, < 1.19.0" "numpy >= 1.16.0"
+
+      # glibc 2.31+ does not have sys/sysctl.h
+      # see https://github.com/tensorflow/tensorflow/issues/45861
+      substituteInPlace third_party/hwloc/BUILD.bazel\
+        --replace "#define HAVE_SYS_SYSCTL_H 1" "#undef HAVE_SYS_SYSCTL_H"
     '';
 
     preConfigure = let

